src/libYARP_sig/include/yarp/sig/PointCloudTypes.h

```diff
@@ -0,0 +1,934 @@
+                                                                    width, precision, curvature);
+                    ret+=tmp;
+                }
+                return ret;
+            }
+            yarp::os::Bottle toBottle()
+            {
+                yarp::os::Bottle ret;
+                ret.addDouble(x);
+                ret.addDouble(y);
+                ret.addDouble(z);
+                ret.addDouble(normal_x);
+                ret.addDouble(normal_y);
+                ret.addDouble(normal_z);
+                ret.addDouble(curvature);
+                return ret;
+
+            }
+            void fromBottle(const yarp::os::Bottle &bt, size_t i)
+            {
+                yarp::os::Bottle* intBt = bt.get(i).asList();
+
+                if (!intBt) {
+                    return;
+                }
+
+                x = static_cast<float>(intBt->get(0).asDouble());
+                y = static_cast<float>(intBt->get(1).asDouble());
+                z = static_cast<float>(intBt->get(2).asDouble());
+                normal_x = static_cast<float>(intBt->get(3).asDouble());
+                normal_y = static_cast<float>(intBt->get(4).asDouble());
+                normal_z = static_cast<float>(intBt->get(5).asDouble());
+                curvature = static_cast<float>(intBt->get(6).asDouble());
+                return;
+            }
+        } XYZ_NORMAL_DATA;
+        YARP_END_PACK
+
+        // point xyz + normals + RGBA
+        YARP_BEGIN_PACK
+        typedef struct
+        {
+            union
+            {
+                float data[4];
+                struct
+                {
+                    float x;
+                    float y;
+                    float z;
+                };
+            };
+            union
+            {
+                float filler_n[4];
+                float normal[3];
+                struct
+                {
+                    float normal_x;
+                    float normal_y;
+                    float normal_z;
+                };
+            };
+            union
+            {
+                struct
+                {
+                    // PCL here uses float rgb, probably for ROS compatibility as stated for PointXYZRGB
+                    // Check compatibility, it should be ok if rgb component are in the right place and we drop 'a'
+                    union
+                    {
+                        struct
+                        {
+                            char b;
+                            char g;
+                            char r;
+                            char a;
+                        };
+                        yarp::os::NetInt32 rgba;
+                    };
+                    float curvature;
+                };
+                float filler_others[4];
+            };
+            yarp::os::ConstString toString(int precision, int width)
+            {
+                yarp::os::ConstString ret = "";
+                char tmp[100];
```

Comment:
(This and all the other **100**s). Why 100? please use a power of 2. Also it is probably worth to define this somewhere)